aten/src/TH/THAllocator.c

```diff
@@ -126,11 +140,67 @@ ptrdiff_t THMapAllocatorContext_size(THMapAllocatorContext *ctx)
 
 void THMapAllocatorContext_free(THMapAllocatorContext *ctx)
 {
-  if (ctx->filename != unknown_filename)
+  if (ctx->filename != unknown_filename) {
     THFree(ctx->filename);
+    THFree(ctx->eventname);
+  }
   THFree(ctx);
 }
 
+#ifdef _WIN32
+typedef struct{
+  HANDLE event;
+  HANDLE handle;
+  void *data;
+  int released;
+  int file;
+} ReleaseContext;
+static DWORD WINAPI WaitForReleaseHandle(LPVOID lpParam) {
+  ReleaseContext *cxt = (ReleaseContext *)lpParam;
+
+  HANDLE event = cxt->event;
+  HANDLE handle = cxt->handle;
+  void *data = cxt->data;
+
+  while (TRUE) {
+    WaitForSingleObject(event, INFINITE);
+
+    printf("force_free\n"); 
+    if (cxt->released) {
+      CloseHandle(event);
```

Comment:
If we comment out L170-171, then there will be no problem but the memory leak continues.
But if we left them here, we will kill the handles that may be unrelated.

Reply:
The possible reason is the Overlapped I/O used in the implementation of `Pipe` in `multiprocessing`. It's asynchronous IO so the handle may be released.